File: program/program/find-gcd-of-two-numbers-by-recursion/FindGcdOfTwoNumbersByRecursion.swift
import Foundation

func gcd(_ a: Int, _ b: Int) -> Int {
    if b == 0 {
        return a
    } else {
        return gcd(b, a % b)
    }
}

// Example usage
let num1 = 12
let num2 = 18
let result = gcd(num1, num2)
print("GCD of \(num1) and \(num2) is \(result)")

File: program/program/find-gcd-of-two-numbers-by-recursion/FindGcdOfTwoNumbersByRecursion.fs
let rec gcd a b =
    if b = 0 then a
    else gcd b (a % b)

// Example usage
let num1 = 12
let num2 = 18

let result = gcd num1 num2
printfn "The GCD of %d and %d is: %d" num1 num2 result
